src/querydb/store.rs

```diff
@@ -0,0 +1,220 @@
+use crate::querydb::prelude::*;
+use pallas::{
+    crypto::hash::Hash,
+    ledger::traverse::{MultiEraBlock, MultiEraPolicyAssets, MultiEraTx},
+};
+use redb::{
+    Database, MultimapTable, ReadOnlyMultimapTable, ReadOnlyTable, ReadTransaction,
+    ReadableMultimapTable, ReadableTable, Table, WriteTransaction,
+};
+use std::{ops::Deref, path::Path};
+
+pub struct Store {
+    inner_store: Database,
+}
+
+impl Store {
+    pub fn new(path: impl AsRef<Path>) -> Result<Self, StoreError> {
+        Ok(Store {
+            inner_store: Database::create(path)
+                .map_err(|e| StoreError::ReDBError(ReDBError::DatabaseInitilization(e)))?,
+        })
+    }
+    pub fn apply_block(&self, block_cbor: &[u8]) -> Result<(), StoreError> {
+        let write_tx: WriteTransaction = self
+            .inner_store
+            .begin_write()
+            .map_err(|e| StoreError::ReDBError(ReDBError::TransactionError(e)))?;
+        let block: MultiEraBlock = Self::store_block(&write_tx, block_cbor)?;
+        Self::store_txs(&write_tx, &block)?;
+        write_tx
+            .commit()
+            .map_err(|e| StoreError::ReDBError(ReDBError::CommitError(e)))?;
+        Ok(())
+    }
+
+    fn store_block<'a>(
+        write_tx: &'a WriteTransaction,
+        block_cbor: &'a [u8],
+    ) -> Result<MultiEraBlock<'a>, StoreError> {
+        let block: MultiEraBlock =
+            MultiEraBlock::decode(block_cbor).map_err(StoreError::BlockDecoding)?;
+        let mut block_table: Table<BlockKeyType, BlockValueType> = write_tx
+            .open_table(BLOCK_TABLE)
+            .map_err(|e| StoreError::ReDBError(ReDBError::TableError(e)))?;
+        let _ = block_table.insert(block.hash().deref(), block_cbor);
+        let mut chain_tip_table: Table<ChainTipKeyType, ChainTipValueType> = write_tx
```

Comment:
Querying the chain tip should be about looking for the latest block, there's no need to have a custom table.

Reply:
I tried doing that in the first place, but the latest block doesn't seem to be so easily retrievable. The [last](https://docs.rs/redb/latest/redb/trait.ReadableTable.html#) method for ReadableTable's returns the key-value pair corresponding to the largest key, and blocks are indexed by their hash (which don't necessarily preserve the order in which they were stored).

I could, for instance, index blocks by a pair (n, block_hash), where n is the insertion order. Would this sound OK?